File: Problemset/power-of-three/power-of-three.cpp

// @Title: 3的幂 (Power of Three)
// @Author: rubychen0611
// @Date: 2021-03-01 10:55:06
// @Runtime: 24 ms
// @Memory: 5.9 MB

class Solution {
public:
    bool isPowerOfThree(int n) {
        return fmod(log10(n) / log10(3), 1) == 0;
    }
};

File: Problemset/power-of-three/power-of-three.java

// @Title: 3的幂 (Power of Three)
// @Author: rubychen0611
// @Date: 2021-08-29 23:36:05
// @Runtime: 15 ms
// @Memory: 38.3 MB

class Solution {
    public boolean isPowerOfThree(int n) {
        if (n <= 0)
            return false;
        while(n > 1)
        {
            if (n % 3 == 0)
                n /= 3;
            else 
                return false;
        }
        return true;
    }
}
